### DISCUSSION TITLE
Adding/ Removing with collection expression

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#8184] [General]
If I want to mutate a collection using another collection, is it considered "good" to add range/remove range using a collection expression?  Or is there potentially an upcoming syntax that will translate to multiple Operation/OperationRange calls?

I'm thinking of Add, Remove, Union, Intersect, etc with the different collections.

**Comment by CyrusNajmabadi** [comment]
> is it considered "good" to add range/remove range using a collection expression?

Can't define 'good', only you can do that :)

> Or is there potentially an upcoming syntax that will translate to multiple Operation/OperationRange calls?

No upcoming syntax.  However, our position on collection-expressions and collection-operations is that the compiler is free to make a lot of assumptions/optimizations there.  Whether or not we will optimize really depends on how much value we see in it and how widespread we think those patterns are.

**Reply by TonyValenti** [reply]
Thanks!

This might be too much magic, but it would be sweet if I could do:

```
collection.Add([ 
  A,
  B,
  .. C,
  .. E.
  F,
]);
```

And have the compiler translate that into Add / AddRange calls that are (likely) more efficient than simply doing ```AddRange(CollectionExpression)```.

It would also be nice if it was pattern based so the compiler would use ```?``` and ```?Range``` as appropriate.

**Reply by FaustVX** [reply]
@TonyValenti 
This already works today, just call `AddRange` instead of `Add`

[sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMBGAWACgsAGAAizwBYBuY4gNwmXIBsoBnBcgXnJgBTAO7kAMlwQAeWAgB8ACgDaeOORxqAzAF0AlPSIduAOgCCAE3MAlCDADmg5dTUBWNQDY9B4lQCcCowQlAD08L2IgA)
